test/xapi/track_test.rb

```diff
@@ -2,6 +2,16 @@
 require_relative '../../lib/xapi'
 
 class TrackTest < Minitest::Test
+  def assert_archive_contains(filenames, zip)
+    files = []
+    Zip::InputStream.open(zip) do |io|
+      while (entry = io.get_next_entry)
+        files << entry.name
+      end
+    end
+    assert_equal filenames.sort, files.sort
+  end
```

Comment:
> Most assert methods use the ordering: expected, actual. What about assert_filenames_in_archive? Too clunky?

You are quite right. `assert_filenames_in_archive` would be fine.

But the more I think about this the more I don't like that method at all.
Here is my reasoning:

The actual assertion we want is: `assert_equal expected, actual`

Where: 
expected = known_filenames.
actual   = filenames_in_the_archive.

Which gives us: `assert_equal known_filenames, filenames_in_the_archive`

We have `known_filenames` which is `files.keys` (line 115 below)

What we need is filenames_in_the_archive

Ideally, we would get this by sending the archive a `filenames` message, (archive.filenames) but since it doesn't respond to that, we can make a wrapper method: `archive_filenames(archive)` which will return an array of filenames.

Extracting that part from our custom assertion, gives us a new method:
```
  def archive_filenames(zip)
    files = []
    Zip::InputStream.open(zip) do |io|
      while (entry = io.get_next_entry)
        files << entry.name
      end
    end
    files
  end
```
Which I like because it is only doing one thing (getting the filenames from the archive) and leaves the assertion (a separate responsibility) as : `assert_equal known_filenames, archive_filenames(archive)`

That done, we can remove the custom `assert_archive_contains/assert_archive_filenames/assert_filenames_in_archive` method completely.